README.md

```diff
@@ -1 +1,109 @@
+
+With Angular, the flow will go like this instead -
+
+![Client-side rendering](http://branchandbound.net/pics/client-side-rendering-only.png)
+
+This looks a lot more complicated, but it isn't! Instead of returning rendered HTML for each user, the server would return the same page for every user. The client will then make a request to the server, requesting what's necessary for that page to render (such as a list of products). The client will then render this page for the user.
+
+This has massive advantages - if 1000 users are on the website, instead of one server rendering 1000 pages, 1000 clients are rendering one page, reducing strain on the server. We can also re-use backend endpoints - our mobile website and our desktop website can both request the same data from the server but display it differently.
+
+## MVC/MVVM
+
+You might have seen the term MVC/MVVM being thrown around. These stand for Model-View-Controller/Model-View-ViewModel respectively. 
+
+These are concepts for how we architecture our software. We do this to promote separation of concerns. Separation of concerns is dividing our code into logical chunks, with each chunk addressing a separate concern. This allows us to create testable, reusable and scalable well designed code.
+
+They are both quite similar, however there are a few differences -
+
+### MVC (Model-View-Controller)
+
+Three parts to MVC; model, view, controller. The model manages the fundamental behaviours and data of the application. This could be retrieved from a database or any other sort of data store.
+
+The view is the user interface. This takes data from the model and displays it nicely for the user. Any user interaction in the view is then handed over to the controller.
+
+The controller receives user input and makes calls to the model and view to update/modify either. This keeps the main aspects of our application separate.
+
+As easy way to visualize this is if we imagine we are ordering a coffee. The Model is the information about the drink we've ordered, the Controller is the barista and the View is the finished drink. The barista would make the drink from our order (the Model), and serve it to us (the View).
+
+So for example, if you walked into a coffee bar and ordered a hot chocolate, you'd get a hot chocolate. If you then wanted marshmallows, you'd ask the barista, who would go back to the till and modify your order to add on marshmallows. They would also then put marshmallows in your drink, and return you the updated drink (the View).
+
+### MVVM (Model-View-ViewModel)
+
+Similar to MVC, however the controller is replaced with a ViewModel.
+
+The ViewModel is responsible for all view logic - changing the model data appropriately for use in the view. (For example, our model might contain a unix timestamp for a date but we need it in human readable format). Your model will contain all of your business logic, and your view is there to display your model (modified by the ViewModel).
```

Comment:
Move this paragraph to directly after you finish your coffee shop example. This will make more sense after seeing the analogy.